docs/design.md

```diff
@@ -834,28 +834,26 @@ overhead of going through Raft.
 Since reads bypass Raft, a new lease holder will, among other things, ascertain
 that its timestamp cache does not report timestamps smaller than the previous
 lease holder's (so that it's compatible with reads which may have occurred on
-the former lease holder). This is accomplished by setting the low water mark of the
-timestamp cache to the expiration of the previous lease plus the maximum clock
-offset.
+the former lease holder). This is accomplished by letting leases enter
+a <i>stasis period</i> (which is just the expiration minus the maximum clock
+ofsset) before the actual expiration of the lease, so that all the next lease
+holder has to do is set the low water mark of the timestamp cache to its
+new lease's start time.
+
+As a lease enters its stasis period, no more reads or writes are served, which
+is undesirable. However, this would only happen in practice if a node became
+unavailable. In almost all practical situations, no unavailability results
+since leases are usually long-lived (and/or eagerly extended, which can avoid
+the stasis period) or proactively transferred away from the lease holder,
+which can also avoid the stasis period by promising not to serve any further
+reads until the next lease goes into effect.
 
 ## Relationship to Raft leadership
 
 The range lease is completely separate from Raft leadership, and so without
-further efforts, Raft leadership and the Range lease may not be represented by the same
-replica most of the time. This is convenient semantically since it decouples
-these two types of leadership and allows the use of Raft as a "black box", but
-for reasons of performance, it is desirable to have both on the same replica.
-Otherwise, sending a command through Raft always incurs the overhead of being
-proposed to the Range lease holder's Raft instance first, which must relay it to the
-Raft leader, which finally commits it into the log and updates its followers,
-including the Range lease holder. This yields correct results but wastes several
-round-trip delays, and so we will make sure that in the vast majority of cases
-Range lease and Raft leadership coincide. A fairly easy method for achieving this is
-to have each new lease period (extension or new) be accompanied by a
-stipulation to the lease holder's replica to start Raft elections (unless it's
-already leading), though some care should be taken that Range lease holdership is
-relatively stable and long-lived to avoid a large number of Raft leadership
-transitions.
+further efforts, Raft leadership and the Range lease may not be represented by the same. Since it's expensive to not have these two roles colocated, each lease
```

Comment:
Can you elaborate on the expense? Something along the lines of `the lease holder needs to forward write operations to the leader`.